sparkle/platform/output.py

```diff
@@ -0,0 +1,287 @@
+from sparkle.configurator.configurator import Configurator
+from sparkle.solver.validator import Validator
+
+from sparkle.CLI.help.nicknames import resolve_object_name
+
+import os
+import json
+from pathlib import Path
+from runrunner.base import Status
+
+
+class ConfigurationPerformance:
+    """Class that stores performance results."""
+    def __init__(self: ConfigurationPerformance, configured_metrics: float,
+                 default_metrics: float) -> None:
+        """Initalize ConfigurationPerformance.
+
+        Args:
+            configured_metrics: The performance of the configured solver
+            default_metrics: The performance result of the default solver
+        """
+        self.configured_metrics = configured_metrics
+        self.default_metrics = default_metrics
+
+
+class ValidationResults:
+    """Class that stores validation information and results."""
+    def __init__(self: ValidationResults, solver: Solver,
+                 configuration: dict, instance_set: InstanceSet,
+                 results: list[list[str, Status, float, float]]) -> None:
+        """Initalize ValidationResults.
+
+        Args:
+            solver: The name of the solver
+            configuration: The configuration being used
+            instance_set: The set of instances
+            results: Validation results in the format:
+                [["instance", "status", "quality", "runtime"]]
+        """
+        self.solver = solver
+        self.configuration = configuration
+        self.instance_set = instance_set
+        self.result_header = ["instance", "status", "quality", "runtime"]
+        self.result_vals = results
+
+
+class ConfigurationResults:
+    """Class that aggregates configuration results."""
+    def __init__(self: ConfigurationResults, performance: ConfigurationPerformance,
+                 results_default: ValidationResults,
+                 results_configured: ValidationResults) -> None:
+        """Initalize ConfigurationResults.
+
+        Args:
+            performance: The performance of the default and configured solver
+            results_default: The default results
+            results_configured: The configured results
+        """
+        self.performance = performance
+        self.configured_results = results_configured
+        self.default_results = results_default
+
+
+class ConfigurationOutput:
+    """Class that collects configuration data and outputs it a json format."""
+    solver: Solver = None
+    configurator: Configurator = None
+    best_config: dict = None
+    training: ConfigurationResults = None
+    test: ConfigurationResults = None
+
+    def __init__(self: ConfigurationOutput, path: Path, solver: Solver,
+                 configurator: Configurator, instance_set_train: InstanceSet,
+                 instance_set_test: InstanceSet,
+                 penalty_multiplier: int, output: Path = None) -> None:
```

Comment:
preferably penalty multiplier is not part of this constructor (too specific) but I can imagine that its currently necessary to make things work. If this is the case, please mark it in the docstring as [To Be Removed]

Reply:
Yes, we discussed this before. Penalty_multiplier is not part of the scenario settings file and therefor needs to be imported separately until that is fixed